pkg/controller/ingress/ingress_controller.go

```diff
@@ -328,6 +339,74 @@ func (r *ReconcileIngress) Reconcile(request reconcile.Request) (reconcile.Resul
 
 }
 
+func (r *ReconcileIngress) verifyAndUpdateASG(instance *extensionsv1beta1.Ingress, remove bool) error {
+	stackName := instance.ObjectMeta.Name
+
+	network, err := r.fetchNetworkingInfo(instance)
+	if err != nil {
+		r.log.Error("error fetching network information", zap.String("stackName", stackName))
+		return err
+	}
+
+	if len(network.ASGNames) == 0 {
+		return nil
+	}
+
+	targetGroupARN, err := cfn.GetResourceID(r.cfnSvc, stackName, "TargetGroup")
+	if err != nil {
+		r.log.Error("error getting TargetGroupARN", zap.String("stackName", stackName))
+		return err
+	}
+
+	for _, asgName := range network.ASGNames {
+		data, err := r.autoscalingSvc.DescribeAutoScalingGroups(&autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: aws.StringSlice([]string{asgName}),
```

Comment:
Same comment as below, if the function takes a slice as input, why send multiple calls

Reply:
- We need to deal with paging (although unlikely for common cases)
- Another verifying one by one to make sure the ASGNames we got from tags are existing.